saleor/graphql/channel/mutations.py

```diff
@@ -51,3 +54,63 @@ class Meta:
         permissions = (ChannelPermissions.MANAGE_CHANNELS,)
         error_type_class = ChannelError
         error_type_field = "channel_errors"
+
+
+class ChannelDeleteInput(graphene.InputObjectType):
+    target_channel = graphene.ID(
+        required=True,
+        description="ID of channel to migrate orders from origin channel.",
+    )
+
+
+class ChannelDelete(ModelDeleteMutation):
+    class Arguments:
+        id = graphene.ID(required=True, description="ID of a channel to delete.")
+        input = ChannelDeleteInput(
+            required=True, description="Fields required to delete a channel."
+        )
+
+    class Meta:
+        description = "Delete a channel."
+        model = models.Channel
+        permissions = (ChannelPermissions.MANAGE_CHANNELS,)
+        error_type_class = ChannelError
+        error_type_field = "channel_errors"
+
+    @classmethod
+    def validate_input(cls, origin_channel_id, target_channel_id):
+        if origin_channel_id == target_channel_id:
+            raise ValidationError(
+                {
+                    "target_channel": ValidationError(
+                        "channelID and targetChannelID cannot be the same. "
+                        "Use different target channel ID.",
+                        code=ChannelErrorCode.CHANNEL_TARGET_ID_MUST_BE_DIFFERENT,
+                    )
+                }
+            )
+
+    @classmethod
+    def perform_delete(cls, origin_channel, target_channel):
+        cls.validate_input(origin_channel, target_channel)
+        cls.migrate_orders_to_target_channel(origin_channel, target_channel)
+        cls.delete_checkouts(origin_channel)
```

Comment:
To make it safer and resilient to race conditions, we could wrap these with `transaction.atomic()` and use `select_for_update` on Order and Checkout querysets inside these two functions. Then we should avoid the risk that new order is created in a channel that will be soon deleted.